### PR TITLE
added cursor modes as https://github.com/ziglibs/ansi-term/issues/15

### PR DESCRIPTION
**Pull request by ZanovelloAlberto**

### CODE REVIEW

File: src/cursor.zig [hunk 1]
```diff
@@ -1,8 +1,23 @@
 const std = @import("std");
-
+const testing = std.testing;
+const fixedBufferStream = std.io.fixedBufferStream;
 const esc = "\x1B";
 const csi = esc ++ "[";
 
+pub const CursorMode = enum(u8) {
+    BLINKING_BLOCK = 1,
```
Comment (joachimschmidt557) [inline src/cursor.zig:8]: The Zig naming convention is that enum values should be camel case, so e.g. `blinking_block`. You can make an exception for the `I` beam (e.g. `blinking_I_beam`).

**Review by joachimschmidt557 — CHANGES_REQUESTED** [review]
Thank you! This is a pretty non-intrusive change, so I'm open to accepting this PR when some small style changes are applied.

**Comment by ZanovelloAlberto** [comment]
it is my first pull request im trying to figure it out who it works, im not sure which button i have to press, anyway i have pushed another commit in my fork

**Comment by joachimschmidt557** [comment]
Yes, pushing commits to the branch connected with this PR will update this PR too. Looks good to me, thanks!

**Pull request merged**